runtime/src/authorize_and_stash.rs

```diff
@@ -100,4 +112,46 @@ impl AuthorizeAndStashCmd {
             Err(CaliptraError::RUNTIME_INSUFFICIENT_MEMORY)
         }
     }
+
+    /// Search for a metadata entry in the sorted `AuthManifestImageMetadataCollection` that matches the firmware ID.
+    ///
+    /// This function performs a binary search on the `image_metadata_list` of the provided `AuthManifestImageMetadataCollection`.
+    /// It compares the firmware ID (`fw_id`) of each metadata entry with the provided `cmd_fw_id_bytes`.
+    ///
+    /// # Arguments
+    ///
+    /// * `auth_manifest_image_metadata_col` - A reference to the `AuthManifestImageMetadataCollection` containing the metadata entries.
+    /// * `cmd_fw_id_bytes` - A reference to a `[u8; 4]` array representing the firmware ID from the command to search for.
+    ///
+    /// # Returns
+    ///
+    /// * `Option<&'a AuthManifestImageMetadata>` - Returns `Some(&AuthManifestImageMetadata)` if a matching entry is found,
+    ///   otherwise returns `None`.
+    ///
+    #[inline(never)]
+    fn find_metadata_entry<'a>(
+        auth_manifest_image_metadata_col: &'a AuthManifestImageMetadataCollection,
+        cmd_fw_id_bytes: &[u8; 4],
+    ) -> Option<&'a AuthManifestImageMetadata> {
+        let mut left = 0;
+        let mut right = auth_manifest_image_metadata_col.entry_count as usize;
+
+        while left < right {
```

Comment:
Can we use `auth_manifest_image_metadata_col.image_metadata_list.binary_search_by_key()`?

Either way, we should add unit tests for this function.

Reply:
Updated.